x-pack/platform/plugins/shared/cases/server/cases_analytics/tasks/scheduler_task/scheduler_task_runner.ts

```diff
@@ -0,0 +1,91 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { type TaskManagerStartContract } from '@kbn/task-manager-plugin/server';
+import type { SavedObjectsClientContract, Logger, ElasticsearchClient } from '@kbn/core/server';
+import type { CancellableTask } from '@kbn/task-manager-plugin/server/task';
+import type { ConfigType } from '../../../config';
+import { getAllSpacesWithCases } from '../../utils';
+import {
+  createCasesAnalyticsIndexesForSpaceId,
+  getIndicesForSpaceId,
+  scheduleCasesAnalyticsSyncTasks,
+} from '../..';
+
+interface SchedulerTaskRunnerFactoryConstructorParams {
+  getUnsecureSavedObjectsClient: () => Promise<SavedObjectsClientContract>;
+  logger: Logger;
+  analyticsConfig: ConfigType['analytics'];
+  getTaskManager: () => Promise<TaskManagerStartContract>;
+  getESClient: () => Promise<ElasticsearchClient>;
+}
+
+export class SchedulerTaskRunner implements CancellableTask {
+  private readonly getUnsecureSavedObjectsClient: () => Promise<SavedObjectsClientContract>;
+  private readonly logger: Logger;
+  private readonly analyticsConfig: ConfigType['analytics'];
+  private readonly getTaskManager: () => Promise<TaskManagerStartContract>;
+  private readonly getESClient: () => Promise<ElasticsearchClient>;
+
+  constructor({
+    getUnsecureSavedObjectsClient,
+    logger,
+    analyticsConfig,
+    getTaskManager,
+    getESClient,
+  }: SchedulerTaskRunnerFactoryConstructorParams) {
+    this.getUnsecureSavedObjectsClient = getUnsecureSavedObjectsClient;
+    this.logger = logger;
+    this.analyticsConfig = analyticsConfig;
+    this.getTaskManager = getTaskManager;
+    this.getESClient = getESClient;
+  }
+
+  public async run() {
+    if (!this.analyticsConfig.index.enabled) {
+      this.logger.debug('Analytics index is disabled, skipping scheduler task.');
+      return;
+    }
+    try {
+      const unsecureSavedObjectsClient = await this.getUnsecureSavedObjectsClient();
+      const spaces = await getAllSpacesWithCases(unsecureSavedObjectsClient);
+      const taskManager = await this.getTaskManager();
+      const esClient = await this.getESClient();
+
+      for (const spaceId of spaces) {
+        const indices = getIndicesForSpaceId(spaceId);
+        const destIndicesExist = await esClient.indices.exists({ index: indices });
+        if (!destIndicesExist) {
+          // Create the necessary analytics indexes without scheduling the sync tasks
+          createCasesAnalyticsIndexesForSpaceId({
+            spaceId,
+            esClient,
+            logger: this.logger,
+            isServerless: false,
+            taskManager,
+          }).catch(() => {
+            this.logger.error(`Failed to create analytics indexes for space ${spaceId}`);
+          });
+        } else {
+          scheduleCasesAnalyticsSyncTasks({
```

Comment:
Could we alternatively have the scheduler do **only**

```
for (const spaceId of spaces) {
    if (!destIndicesExist) {
        createCasesAnalyticsIndexesForSpaceId({...});
        scheduleCasesAnalyticsSyncTasks({...});
    }
}
```

Why periodically try to reschedule for every space and every different possible index?

If the index does not exist, we create it and schedule the task once; no need for constantly making calls to the task manager.

Reply:
One of the reasons for this is to make sure that the task was scheduled successfully. Right now, in case the `sync` task isn't scheduled (e.g. due to some error in `TaskManager`), the sync task will not be scheduled anymore.